src/components/AuthorViewpoint/FactListCard.tsx

```diff
@@ -0,0 +1,116 @@
+"use client";
+import { Fact } from "@/types/conversations.types";
+import EditViewpointFact from "@/components/AuthorViewpoint/EditViewpointFact";
+import { MagnifyingGlassIcon, PlusIcon } from "@heroicons/react/24/outline";
+import { Select, Button } from "@mantine/core";
+import { useState, useEffect } from "react";
+import { allFacts } from "@/mock/conversationMock";
+
+type FactListCardProps = {
+    viewpointFactList: Fact[];
+    setViewpointFactList: (FactList: Fact[]) => void;
+};
+
+export default function FactListCard({
+    viewpointFactList,
+    setViewpointFactList,
+}: FactListCardProps) {
+    const [searchData, setSearchData] = useState<Fact[]>(allFacts); // eslint-disable-line
+    const [selectedFactId, setSelectedFactId] = useState<string>("");
+    const [searchValue, setSearchValue] = useState<string>(""); // eslint-disable-line
+
+    useEffect(() => {
+        const selectedFact = searchData.find(
+            (fact) => String(fact.id) == selectedFactId,
+        );
+        if (selectedFact) {
+            //move the selected fact to the viewpointFactList
+            setViewpointFactList([...viewpointFactList, selectedFact]);
+
+            //remove the selected fact from the searchData
+            const newSearchData = searchData.filter(
+                (fact) => fact.id !== selectedFact.id,
+            );
+            setSearchData(newSearchData);
+        }
+    }, [selectedFactId, setSelectedFactId]);
+
+    const removeFact = (factId: string) => {
+        //remove the fact from the viewpointFactList
+        const removedFact = viewpointFactList.find(
+            (fact) => String(fact.id) == factId,
+        );
+        const newSelectedFacts = viewpointFactList.filter(
+            (fact) => String(fact.id) !== factId,
+        );
+        setViewpointFactList(newSelectedFacts);
```

Comment:
The same applies here, consider basing filters on data passed in by the setState hook instead of relying on the state variable